File: 1863_sum_of_all_subset_xor_totals.java
class Solution {
    int s = 0;
    public int subsetXORSum(int[] x) {
        go(0, 0, x);
        return s;
    }
    void go(int i, int t, int[] x) {
        if (i == x.length) s += t;
        else { go(i + 1, t, x); go(i + 1, t ^ x[i], x); }
    }
}

File: 1863_sum_of_all_subset_xor_totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& z, int a = 0, int b = 0) {
        return a == z.size() ? b : subsetXORSum(z, a + 1, b) + subsetXORSum(z, a + 1, b ^ z[a]);
    }
};
